akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HttpCookieTest.scala

```diff
@@ -0,0 +1,79 @@
+import akka.http.scaladsl.model.DateTime
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+
+class HttpCookieTest extends AnyFreeSpec with Matchers {
+  "HttpCookies" - {
+    "when rendering it should" - {
+      "include name and value" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue")
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must include("cookieName=cookieValue")
+      }
+      "include Expires if present" in {
+        val expires = DateTime.now
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue", expires = Some(expires))
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must include(s"; Expires=${expires.toRfc1123DateTimeString}")
+      }
+      "include Max-Age if present" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue", maxAge = Some(101))
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must include("; Max-Age=101")
+      }
+      "include Domain if present" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue", domain = Some("akka.io"))
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must include("; Domain=akka.io")
+      }
+      "include Path if present" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue", path = Some("/akka-http"))
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must include("; Path=/akka-http")
+      }
+      "include Secure if cookie is secure" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue", secure = true)
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must include("; Secure")
+      }
+      "not include Secure if cookie is not secure" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue", secure = false)
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must not include ("; Secure")
+      }
+      "include HttpOnly if cookie is http only" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue", httpOnly = true)
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must include("; HttpOnly")
+      }
+      "not include HttpOnly if cookie is not http only" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue", httpOnly = false)
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must not include ("; HttpOnly")
+      }
+      "not include SameSite if not specified" in {
+        val cookie = HttpCookie(name = "cookieName", value = "cookieValue")
+        val rendering = new StringRendering()
+        cookie.render(rendering).get must not include ("; SameSite=None")
```

Comment:
Since we had it at another place: we need to verify if parsing should be case-insensitive and if yes add tests that make sure it works for whatever casing.